core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -162,6 +169,17 @@ public Object getFieldValue(MasterWorkerInfo o) {
   private final IndexedSet<MasterWorkerInfo> mLostWorkers =
       new IndexedSet<>(ID_INDEX, ADDRESS_INDEX);
 
+  /** Listeners to call when lost workers are found. */
+  private final BlockingQueue<Consumer<Address>> mLostWorkerFoundListeners
+      = new LinkedBlockingQueue<>();
+
+  /** Listeners to call when workers are lost. */
+  private final BlockingQueue<Consumer<Address>> mWorkerLostListeners = new LinkedBlockingQueue<>();
```

Comment:
I think `List/ArrayList` or `Set/HashSet` should be fine for all of these, we don't want to block